### PR TITLE
Fix NRE in nullable walker

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn/issues/61013

### CODE REVIEW

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs [hunk 1]
```diff
@@ -7880,7 +7880,7 @@ void reportBadDelegateParameter(BindingDiagnosticBag bag, MethodSymbol sourceInv
                     {
                         var delegateType = targetType.GetDelegateType();
                         VisitLambda(lambda, delegateType, stateForLambda);
-                        if (reportRemainingWarnings)
+                        if (reportRemainingWarnings && delegateType is not null)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs:7883]: >delegateType is not null [](http://example.com/codeflow?start=55&length=24)

Thanks. Perhaps adding `nullable enable` around `case ConversionKind.AnonymousFunction` as well.

File: src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs [hunk 2]
```diff
@@ -6782,5 +6782,41 @@ static void Main()
             Assert.Equal(RefKind.In, lambdas[1].Parameters[0].RefKind);
             Assert.Equal(RefKind.Out, lambdas[2].Parameters[0].RefKind);
         }
+
+        [Fact]
+        public void InvalidCast()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs:6787]: >public [](http://example.com/codeflow?start=8&length=6)

Consider adding a `[WorkItem]` attribute.

File: src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs [hunk 3]
```diff
@@ -6782,5 +6782,41 @@ static void Main()
             Assert.Equal(RefKind.In, lambdas[1].Parameters[0].RefKind);
             Assert.Equal(RefKind.Out, lambdas[2].Parameters[0].RefKind);
         }
+
+        [Fact]
```
Comment (Youssef1313) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs:6786]:
```suggestion
        [WorkItem(61013, "https://github.com/dotnet/roslyn/issues/61013")]
        [Fact]
```

**Review by cston — APPROVED** [review]

**Comment by Youssef1313** [comment]
@333fred auto-merge is stuck. Can you manually merge? Thanks!

**Comment by 333fred** [comment]
Thanks @Youssef1313!

**Pull request merged**